validator/sawtooth_validator/networking/interconnect.py

```diff
@@ -463,21 +463,32 @@ def __init__(self,
     @property
     def connection_id(self):
         return hashlib.sha512(
-                self._send_receive_thread.connection.encode()).hexdigest()
+            self._send_receive_thread.connection.encode()).hexdigest()
 
-    def send(self, message_type, data):
-        """
-        Send a message of message_type
-        :param message_type: validator_pb2.Message.* enum value
-        :param data: bytes serialized protobuf
-        :return: future.Future
+    def send(self, message_type, data, callback=None):
+        """Sends a message of message_type
+
+        Args:
+            message_type (validator_pb2.Message): enum value
+            data (bytes): serialized protobuf
+            callback (function): a callback function to call when a
+                response to this message is received
+
+        Returns:
+            future.Future
         """
         message = validator_pb2.Message(
             correlation_id=_generate_id(),
             content=data,
             message_type=message_type)
 
-        fut = future.Future(message.correlation_id, message.content)
+        fut = future.Future(message.correlation_id, message.content,
+                            has_callback=True if callback is not None
```

Comment:
Seems like we could just pass the callback here, and save a step, and some possible mismatch of `has_callback=True`, with no callback added.